lib/babe/verify.go

```diff
@@ -128,45 +128,44 @@ func (v *VerificationManager) SetOnDisabled(index uint32, header *types.Header)
 // It checks the next epoch and config data stored in memory only if it cannot retrieve the data from database
 // It returns an error if the block is invalid.
 func (v *VerificationManager) VerifyBlock(header *types.Header) error {
-	var (
-		info *verifierInfo
-		has  bool
-	)
+	parentHeader, err := v.blockState.GetHeader(header.ParentHash)
+	if err != nil {
+		return fmt.Errorf("getting header: %w", err)
+	}
 
 	epoch, err := v.epochState.GetEpochForBlock(header)
 	if err != nil {
-		return fmt.Errorf("failed to get epoch for block header: %w", err)
+		return fmt.Errorf("getting epoch for block header: %w", err)
 	}
 
-	v.lock.Lock()
+	verifierInfoEpoch := epoch
```

Comment:
why not just assign to variable called `verifierInfoEpoch`?

Reply:
We still need this extra variable I changed its name to `epochWhereDataDescriptorIs`, it will be initialized holding the epoch of the current block, and once we identify a epoch skip we will change its value to `parentEpoch + 1` but we keep holding the current block epoch since to verify the block we need it